pkg/operator/connectivitycheckcontroller/connectivity_check_controller.go

```diff
@@ -0,0 +1,191 @@
+	"github.com/openshift/library-go/pkg/operator/events"
+	"github.com/openshift/library-go/pkg/operator/resource/resourcehelper"
+	"github.com/openshift/library-go/pkg/operator/v1helpers"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/client-go/kubernetes"
+	corev1listers "k8s.io/client-go/listers/core/v1"
+
+	"github.com/openshift/cluster-kube-apiserver-operator/pkg/operator/operatorclient"
+)
+
+type ConnectivityCheckController interface {
+	factory.Controller
+}
+
+func NewConnectivityCheckController(
+	kubeClient kubernetes.Interface,
+	operatorClient v1helpers.StaticPodOperatorClient,
+	kubeInformersForNamespaces v1helpers.KubeInformersForNamespaces,
+	operatorcontrolplaneClient *operatorcontrolplaneclient.Clientset,
+	recorder events.Recorder,
+) ConnectivityCheckController {
+	c := &connectivityCheckController{
+		kubeClient:                 kubeClient,
+		operatorClient:             operatorClient,
+		operatorcontrolplaneClient: operatorcontrolplaneClient,
+		endpointsLister:            kubeInformersForNamespaces.InformersFor("openshift-apiserver").Core().V1().Endpoints().Lister(),
+		serviceLister:              kubeInformersForNamespaces.InformersFor("openshift-apiserver").Core().V1().Services().Lister(),
+	}
+	c.Controller = factory.New().
+		WithSync(c.Sync).
+		WithInformers(
+			operatorClient.Informer(),
+			kubeInformersForNamespaces.InformersFor("openshift-apiserver").Core().V1().Endpoints().Informer(),
+			kubeInformersForNamespaces.InformersFor("openshift-apiserver").Core().V1().Services().Informer(),
+		).
+		ToController("ConnectivityCheckController", recorder.WithComponentSuffix("connectivity-check-controller"))
+	return c
+}
+
+type connectivityCheckController struct {
+	factory.Controller
+	kubeClient                 kubernetes.Interface
+	operatorClient             v1helpers.StaticPodOperatorClient
+	operatorcontrolplaneClient *operatorcontrolplaneclient.Clientset
+	endpointsLister            corev1listers.EndpointsLister
+	serviceLister              corev1listers.ServiceLister
+}
+
+func (c *connectivityCheckController) Sync(ctx context.Context, syncContext factory.SyncContext) error {
+	operatorSpec, _, _, err := c.operatorClient.GetStaticPodOperatorState()
+	if err != nil {
+		return err
+	}
+	switch operatorSpec.ManagementState {
+	case operatorv1.Managed:
+	case operatorv1.Unmanaged:
+		return nil
+	case operatorv1.Removed:
+		return nil
+	default:
+		syncContext.Recorder().Warningf("ManagementStateUnknown", "Unrecognized operator management state %q", operatorSpec.ManagementState)
+		return nil
+	}
+	managePodNetworkConnectivityChecks(ctx, c.kubeClient, c.operatorcontrolplaneClient, operatorSpec, c.endpointsLister, c.serviceLister, syncContext.Recorder())
+	return nil
+}
+
+func managePodNetworkConnectivityChecks(ctx context.Context, client kubernetes.Interface,
```

Comment:
I don't see where you're cleaning up if a master is removed.

You can make a bug and fix it post-merge

Reply:
> You can make a bug and fix it post-merge

and when you do, you'll need a test